component/newstorage/couchdb/store.go

```diff
@@ -359,11 +380,66 @@ func (s *Store) GetBulk(keys ...string) ([][]byte, error) {
 // If TagValue is not provided, then all data associated with the TagName will be returned.
 // For now, expression can only be a single tag Name + Value pair.
 // If no options are provided, then defaults will be used.
+// For improved performance, ensure that the tag name you are querying is included in the store config, as this
+// will ensure that it's indexed in CouchDB.
+// TODO (#44) Should we make the store config mandatory?
 func (s *Store) Query(expression string, options ...newstorage.QueryOption) (newstorage.Iterator, error) {
-	return &couchDBResultsIterator{}, errors.New("not implemented")
+	if expression == "" {
+		return &couchDBResultsIterator{}, errInvalidQueryExpressionFormat
+	}
+
+	queryOptions := getQueryOptions(options)
+
+	expressionSplit := strings.Split(expression, ":")
+
+	switch len(expressionSplit) {
+	case expressionTagNameOnlyLength:
+		expressionTagName := expressionSplit[0]
+
+		findQuery := fmt.Sprintf(`{"selector":{"%s":{"$exists":true}},"limit":%d}`,
+			expressionTagName, queryOptions.PageSize)
+
+		resultRows, err := s.db.Find(context.Background(), findQuery)
+		if err != nil {
+			return nil, fmt.Errorf(failSendRequestToFindEndpoint, err)
+		}
+
+		queryWithPageSizeAndBookmarkPlaceholders := `{"selector":{"` +
```

Comment:
Above you are using Sprintf for generating the query and here you're using concatenation. Should be consistent.

Reply:
I had to use concatenation for the "placeholders" once since it itself actually contains formatting directives that I need passed into the iterator. The iterator has to generate new queries dynamically after the initial one for paging.